scipy/stats/stats.py

```diff
@@ -7234,7 +7234,7 @@ def ks_2samp(data1, data2, alternative='two-sided', mode='auto'):
         mode = 'exact' if max(n1, n2) <= MAX_AUTO_N else 'asymp'
     elif mode == 'exact':
         # If lcm(n1, n2) is too big, switch from exact to asymp
-        if n1g >= np.iinfo(np.int_).max / n2g:
+        if n1g >= np.iinfo(np.int32).max / n2g:
```

Comment:
Looks like the change @rgommers requested--I'll mark as approved but let them merge it in case something was going to get checked in CI for example.